sockeye/rnn_attention.py

```diff
@@ -49,7 +52,8 @@ def __init__(self,
                  query_num_hidden: int,
                  layer_normalization: bool,
                  config_coverage: Optional[coverage.CoverageConfig] = None,
-                 num_heads: Optional[int] = None) -> None:
+                 num_heads: Optional[int] = None,
+                 dtype: Optional[str] = C.DTYPE_FP32) -> None:
```

Comment:
is `dtype` indeed `Optional`, meaning it can be `None` given that you have a default `str` argument? You use it below in `np.finfo`, which defaults to `float64` when given None on my system.

Reply:
fixed